conda/cli/conda_argparse.py

```diff
@@ -126,18 +136,26 @@ def do_call(args: argparse.Namespace, parser: ArgumentParser):
     """
     # let's see if during the parsing phase it was discovered that the
     # called command was in fact a plugin subcommand
-    plugin_subcommand = getattr(args, "plugin_subcommand", None)
-
-    if plugin_subcommand:
+    if plugin := getattr(args, "_plugin_subcommand", None):
         # pass on the rest of the plugin specific args or fall back to
         # the whole discovered arguments
-        try:
-            plugin_args = tuple(args.plugin_args)
-        except AttributeError:
-            plugin_args = args
-        context.plugin_manager.invoke_pre_commands(plugin_subcommand.name)
-        result = plugin_subcommand.action(plugin_args)
-        context.plugin_manager.invoke_post_commands(plugin_subcommand.name)
+        context.plugin_manager.invoke_pre_commands(plugin.name)
+        result = plugin.action(getattr(args, "_args", args))
+        context.plugin_manager.invoke_post_commands(plugin.name)
+    elif name := getattr(args, "_executable", None):
+        # run the subcommand from executables; legacy path
+        deprecated.topic(
+            "23.3",
+            "23.9",
+            topic="Loading conda subcommands via executables",
+            addendum="Use the plugin system instead.",
+        )
+        executable = find_executable(f"conda-{name}")
+        if not executable:
+            from ..exceptions import CommandNotFoundError
+
+            raise CommandNotFoundError(name)
+        _exec([executable, *args._args], os.environ)
```

Comment:
```suggestion
        return _exec([executable, *args._args], os.environ)
```

Could we put a return statement here just to stress that this call effectively ends the function operation? (This becomes clear once you examine the call stack, but it would be nice to it have there anyways).

Reply:
I'm indifferent to this but can see how doing this could help with readability, added in https://github.com/conda/conda/pull/12910/commits/bfd9801d9fedaff1783146ccc13fc1656cfda0ab